src/NonBlocking/ConcurrentDictionary/DictionaryImpl`2.cs

```diff
@@ -14,11 +14,8 @@ namespace NonBlocking
     internal abstract class DictionaryImpl<TKey, TValue>
         : DictionaryImpl
     {
-        internal readonly bool valueIsValueType = typeof(TValue).IsValueType;
```

Comment:
I have tried removing `valueIsValueType` in the past and it did not result in measurable improvements. On the other hand the runtimes that do not guarantee that `IsValueType` is an intrinsic might still be around and without that it is a pretty expensive call. 

I'd prefer to keep the `valueIsValueType` if it does not cost too much or at least hide this behind some kind of `#ifdef`. 
If the gain is small, it is better not to accidentally ruin the perf for someone for whom IsValueType is not an intrinsic. 

Did this yield a noticeable improvement?

Reply:
.NET 6+ is able to fold this, however, just caching into a static readonly field would cause NativeAOT to have cctor init check, and keeping it in an instance field would prevent elimination of the dead code, keeping the branch as is.

I don't think performance on Mono or .NET Framework is as important (especially with Mono planning to complete migration to vanilla .NET some time next year), as present day and future targets. Sadly this can't be properly `#ifdef`'d because the only targets we can rely on are `NS2.0` and `NS2.1`.

Reply:
Let me split this on `NS2.0` and `NS2.1`.